src/engraving/dom/undo.cpp

```diff
@@ -3298,3 +3298,59 @@ void ChangeSpanArpeggio::flip(EditData*)
     m_chord->setSpanArpeggio(m_spanArpeggio);
     m_spanArpeggio = f_spanArp;
 }
+
+AddSystemLock::AddSystemLock(const SystemLock* systemLock)
+    : m_systemLock(systemLock) {}
+
+void AddSystemLock::undo(EditData*)
+{
+    Score* score = m_systemLock->startMB()->score();
+    score->removeSystemLock(m_systemLock);
+}
+
+void AddSystemLock::redo(EditData*)
+{
+    Score* score = m_systemLock->startMB()->score();
+    score->addSystemLock(m_systemLock);
+}
+
+void AddSystemLock::cleanup(bool undo)
+{
+    if (!undo) {
+        delete m_systemLock;
+        m_systemLock = 0;
```

Comment:
Let's always prefer `nullptr` over `0` in new code
```suggestion
        m_systemLock = nullptr;
```
(same below)